packages/default-theme/src/logic/useDomains.js

```diff
@@ -0,0 +1,39 @@
+import { computed } from "@vue/composition-api"
+import { getApplicationContext } from "@shopware-pwa/composables"
+
+export const useDomains = (rootContext) => {
+  const { router, routing } = getApplicationContext(rootContext, "useDomains")
+
+  const availableDomains = computed(() => routing.availableDomains || [])
+  const currentDomainId = computed(() => routing.getCurrentDomain().domainId)
+
+  const changeDomain = async (domainId) => {
+    if (domainId === currentDomainId.value) return
+
+    const domainForLocaleFound = routing.availableDomains.find(
+      (domain) => domain.domainId == domainId
+    )
+
+    if (!domainForLocaleFound) {
+      return
+    }
+
+    // remove domain
+    const cleanPath = rootContext.$route.fullPath.replace(
+      routing.getCurrentDomain().url,
+      ""
+    )
+    // build the new url based on current domain
```

Comment:
So right here we're stripping out the old domain from the path and replace it with the new domain. However, we can't be sure that this new path is a valid resource.

This is the place where we should check, what the correct path for the new domain is.

Reply:
For static pages the way given works, for page-resolver pages, we have to do the following:

```javascript
// POST http://localhost/store-api/v3/seo-url
// 
// Headers
//
// sw-language-id	2fbb5fe2e29a4d70aa5854ce7ce3e20b // New language id 

{
	"filter": [{
		"type": "equals",
		"field": "foreignKey",
		"value": "0096b08cf3524e29a6a4275988e45643"
	}],
	"includes": {
		"seo_url": ["seoPathInfo"]
	}
}
```